src/dcd_mapping/vrs_map.py

```diff
@@ -348,7 +354,7 @@ def _map_protein_coding_pro(
         False,
     )
 
-    if pre_mapped_protein and post_mapped_protein:
+    if pre_mapped_protein:
```

Comment:
We know this on 344, so we could just check pre_mapped_protein right after constructing it and return None without attempting to build the post mapped protein that we won't need. (I'm sort of sympathetic to the argument the code looks nicer the way it is though).

```
pre_mapped_protein = _construct_vrs_allele

if pre_mapped_protein is None:
      return None

post_mapped_protein = _construct_vrs_allele
return mapped score
```